openarc/openarc-ar.c

```diff
@@ -491,6 +491,14 @@ ares_parse(u_char *hdr, struct authres *ar)
 
 			r = 0;
 
+			if (strcasecmp((char *) tokens[c], "none") == 0)
+			{
+				if (n > 0)
+					n--;
+
+				continue;
```

Comment:
Seems to me this should change "state" to some new terminal state to be sure nothing follows it.